lib/active_model/serializer.rb

```diff
@@ -164,6 +164,14 @@ def json_key
       end
     end
 
+    def id
+      object.id if object
+    end
+
+    def type
+      object.class.to_s.demodulize.underscore.pluralize
```

Comment:
Good idea, I was already planning on adding it to the documentation.

However, we need to figure out what to do about the `id` here. The main issue is that the jsonapi adapter requires an id attribute, so how do we handle the cases where the user doesn't add the `id` attribute to the configuration (or any other required attribute)? Raise an error, or add the `id` transparently? If the latter, I can implement a generic solution, something like

```
    def attributes(options = {})
      attributes =
        if options[:fields]
          self.class._attributes & options[:fields]
        else
          self.class._attributes.dup
        end

      attributes += options[:required_fields] if options[:required_fields]

      attributes.each_with_object({}) do |name, hash|
        hash[name] = read_attribute(name)
      end
    end

    def read_attribute(attr)
      if respond_to?(attr)
        send(attr)
      else
        object.read_attribute_for_serialization(attr)
      end
    end
```